drivers/hwinfo/hwinfo_imxrt.c

```diff
@@ -4,6 +4,8 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
+#define DT_DRV_COMPAT nxp_imx_src
```

Comment:
If the existing driver works _without_ requiring the SRC to be present, I think this functionality needs to go in as a different (dedicated) driver (unless all i.MX RT have the SRC IP, in which case I think it should be enabled on all those SoCs within this PR).

Device ID and reset cause does not need to be provided by a single HWINFO driver; see for instance the `CONFIG_HWINFO_MCUX_SIM` and `CONFIG_HWINFO_MCUX_RCM` drivers. Both the SIM and RCM are present on e.g. the NXP KE1xF, one providing a unique ID, the other a reset cause.

Reply:
Thanks for the suggestion, @henrikbrixandersen 

I've moved the driver to a separate file, although I'm not entirely sure of the naming (`mcux/imx/rt`), maybe someone from NXP could suggest more appropriate file/config names if needed.